src/zenml/new/pipelines/build_utils.py

```diff
@@ -585,3 +532,32 @@ def verify_custom_build(
             "your local machine or the image tags have been "
             "overwritten since the original build happened."
         )
+
+
+def compute_stack_checksum(stack: StackResponse) -> str:
+    """Compute a stack checksum.
+
+    Args:
+        stack: The stack for which to compute the checksum.
+
+    Returns:
+        The checksum.
+    """
+    hash_ = hashlib.md5()  # nosec
+
+    # This checksum is used to see if the stack has been updated since a build
+    # was created for it. We create this checksum not with specific requirements
+    # as these might change with new ZenML releases, but they don't actually
+    # invalidate those Docker images.
+    required_integrations = sorted(
```

Comment:
Can any `custom` flavors lead to weird cases here?

Reply:
Also, any change in the configuration of the component can lead to problems when executing runs even though the required integrations remain the same. 

Perhaps we can use the `updated` field of the stack and compare it to the `created` field of the run template and throw a warning if the stack is updated after the creation of the run template.

Reply:
Custom flavors have `integration=None` which gets filtered out. This is just for the checksum computation however, running templates from the server does extra checks that prevent custom flavors completely, so we don't have to worry about it.

Reply:
As for the configuration update: You're right that it might lead to problems, however I think in most cases it actually doesn't. Changing the experiment name of an experiment tracker, adding labels to a container registry, ... . So while my implementation here is not perfect, it definitely covers cases that will always fail (new integrations in the stack that are not built into the Docker image). IMO we should go with this, and solve the corner cases by introducing real stack versioning. WDYT?

Reply:
Sure, the implementation is quite nice right now. My proposal was not to make it fail with the misalignment between the `updated` and `created` fields but rather to throw just a warning. 

Also, you are right. The ultimate solution here is the stack versioning. Until that is implemented, I think this is a good interim solution. 👍